src/overlay.py

```diff
@@ -63,6 +65,18 @@ def add_attribute(
         attribute_format_as: type,
     ) -> None:
         """Add an attribute to the overlay. The attribute can be formatted as int, float, or str."""
+
+        if not hasattr(obj, attribute_name):
+            if isinstance(obj, LinearPhysicalObject) and not hasattr(
```

Comment:
Does this inner nested if have any effect? I do not see it tbh.
In case it is indeed useful, sth like
```python
if condition:
  next_line = "blubb"
elif condition2:
  next_line ="bla"
else:
  next_line = "ok if you insist"
self.line_order.append(next_line)
```
with only one exit from the function would be less confusing to read

Reply:
wrt to this difficult to understand construction I would even more like to have the more generic approach directly referring to `obj.kinematic.velocity` instead of `obj.velocity`. If we want to build shortcuts, probably another place in the code would be more suitable.

Removing nested layers of `if` clauses is almost always an improvement, well done (but still consider simplifying by kicking the special treatment of `kinematic.` attributes